src/ng/compile.js

```diff
@@ -1877,6 +1877,36 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
         return value || null;
       }
 
+      function setupControllers($element, attrs, transcludeFn, controllerDirectives, isolateScope, scope) {
+        var elementControllers = createMap();
+        for (var controllerKey in controllerDirectives) {
+          var directive = controllerDirectives[controllerKey];
+          var locals = {
+            $scope: directive === newIsolateScopeDirective || directive.$$isolateScope ? isolateScope : scope,
```

Comment:
But that doesn't mean `directive === newIsolateScopeDirective`, there might be other controllers that don't want the isolated scope.  Unless `directive.$$isolateScope` will always be true on that case? (which I'm assuming it is not)